Unix/http/http.c

```diff
@@ -816,13 +817,20 @@ static Http_CallbackResult _WriteHeader(
 
     if (handler->sendPage)
     {
-        buf_size = (size_t)Snprintf(
-            currentLine,
-            sizeof(currentLineBuff),
-            RESPONSE_HEADER_FMT,
-            (int)handler->httpErrorCode,
-            _GetHttpErrorCodeDescription(handler->httpErrorCode),
-            (int)handler->sendPage->u.s.size );
+        if (handler->pSendAuthHeader)
+        {
+            buf_size = (size_t)Snprintf( currentLine, sizeof(currentLineBuff), RESPONSE_HEADER_FMT,
+                                         (int)handler->httpErrorCode, _GetHttpErrorCodeDescription(handler->httpErrorCode),
+                                         (int)handler->sendPage->u.s.size, handler->pSendAuthHeader );
+            PAL_Free(handler->pSendAuthHeader);
+            handler->pSendAuthHeader = NULL;
+        }
+        else 
+        {
+            buf_size = (size_t)Snprintf( currentLine, sizeof(currentLineBuff), RESPONSE_HEADER_FMT,
+                                         (int)handler->httpErrorCode, _GetHttpErrorCodeDescription(handler->httpErrorCode),
+                                         (int)handler->sendPage->u.s.size, "" );
```

Comment:
isn't this going to add an extra blank line to the headers, which actually means we have a blank line at the beginning of the body which is going to mean the content length is going to be off by 2?
Maybe you need two versions of the RESPONSE_HEADER_FMT.
As a secondary note, sprint is one of the slowest functions known to developers. Only sleep is slower :) Maybe we should use memcpy where possible.